tap_zendesk/client.py

```diff
@@ -24,127 +30,184 @@
 
-        Returns:
-            A dictionary of HTTP headers.
-        """
-        headers = {}
-        if "user_agent" in self.config:
-            headers["User-Agent"] = self.config.get("user_agent")
-        # If not using an authenticator, you may also provide inline auth headers:
-        # headers["Private-Token"] = self.config.get("auth_token")  # noqa: ERA001
-        return headers
+    def get_start_time(self, context: dict | None) -> int:
+        """Get the start time for the initial incremental export."""
+        replication_key_value = self.get_starting_replication_key_value(context)
+        if replication_key_value:
+            # Parse the string to a datetime object
+            record_date = datetime.fromisoformat(replication_key_value)
+            start_time = int(record_date.timestamp())
+        else:
+            start_time = int(time.time()) - 86400  # 24 hours ago as a default
+        return start_time
+
+    def parse_response(self, response: requests.Response) -> Iterable[dict]:
+        """Parse the response and return an iterator of result records."""
+        if not response.content:
+            raise ValueError(f"Received empty response for URL: {response.url}")
 
-    def get_new_paginator(self) -> BaseAPIPaginator:
-        """Create a new pagination helper instance.
+        response_json = response.json()
+        self.logger.debug(f"Response JSON: {response_json}")
 
-        If the source API can make use of the `next_page_token_jsonpath`
-        attribute, or it contains a `X-Next-Page` header in the response
-        then you can remove this method.
+        end_date_str = self.config.get("end_date")
+        end_date = datetime.fromisoformat(end_date_str) if end_date_str else None
 
-        If you need custom pagination that uses page numbers, "next" links, or
-        other approaches, please read the guide: https://sdk.meltano.com/en/v0.25.0/guides/pagination-classes.html.
+        if end_date and end_date.tzinfo is None:
+            end_date = end_date.replace(tzinfo=timezone.utc)
 
-        Returns:
-            A pagination helper instance.
-        """
-        return super().get_new_paginator()
+        replication_key = self.replication_key
 
-    def get_url_params(
+        for record in extract_jsonpath(self.records_jsonpath, input=response_json):
+            if not record:
+                self.logger.error("Received empty record")
+                continue
+
+            record_date_str = record.get(replication_key)
+            if record_date_str:
+                record_date = datetime.fromisoformat(record_date_str)
+                if record_date.tzinfo is None:
+                    record_date = record_date.replace(tzinfo=timezone.utc)
+                if end_date and record_date > end_date:
+                    self.logger.info(
+                        f"Stopping data fetch as record date {record_date} exceeds end date {end_date}")
+                    return
```

Comment:
nit: Maybe it's better to extract this check to a method to make it more readable.
```suggestion
            if end_date:
              record_date_str = record.get(replication_key)
              if record_date_str:
                  record_date = datetime.fromisoformat(record_date_str)
                  if record_date.tzinfo is None:
                      record_date = record_date.replace(tzinfo=timezone.utc)
                  if record_date > end_date:
                      self.logger.info(
                          f"Stopping data fetch as record date {record_date} exceeds end date {end_date}")
                      return
```

Reply:
Thank you! Done in: [4c10319](https://github.com/Automattic/tap-zendesk-singer-sdk/pull/1/commits/4c10319be06ad3485e2d8ee6fd72c34ac7455948)